pkgs/by-name/gh/ghostty/package.nix

```
@@ -0,0 +1,138 @@
+  glib,
+  gsettings-desktop-schemas,
+  gtk4,
+  harfbuzz,
+  libadwaita,
+  libGL,
+  libpng,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  ncurses,
+  nix-update-script,
+  oniguruma,
+  pandoc,
+  pkg-config,
+  wrapGAppsHook4,
+  zig_0_13,
+  zlib,
+}:
+let
+  # Ghostty needs to be built with --release=fast, --release=debug and
+  # --release=safe enable too many runtime safety checks.
+  zig_hook = zig_0_13.hook.overrideAttrs {
+    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+  };
+in
+stdenv.mkDerivation (final: {
+  pname = "ghostty";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ghostty-org";
+    repo = "ghostty";
+    rev = "v${final.version}";
+    hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
+  };
+
+  nativeBuildInputs = [
+    git
+    ncurses
+    pandoc
+    pkg-config
+    wrapGAppsHook4
+    zig_hook
+  ];
+
+  buildInputs = [
+    bzip2
+    expat
+    fontconfig
+    freetype
+    glib
+    gsettings-desktop-schemas
+    gtk4
+    harfbuzz
+    libadwaita
+    libGL
+    libpng
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    oniguruma
+    zlib
+  ];
+
+  dontPatch = true;
+  dontConfigure = true;
+  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
+  doCheck = false;
+
+  deps = callPackage ./deps.nix { name = "${final.pname}-cache-${final.version}"; };
+
+  zigBuildFlags = [
+    "--system"
+    "${final.deps}"
+    "-Dversion-string=${final.version}"
+  ];
+
+  outputs = [
+    "out"
+    "terminfo"
+    "shell_integration"
+    "vim"
+  ];
+
+  checkPhase = ''
+    zig build --verbose -Dcpu=baseline -Doptimize=Debug ${lib.strings.concatStringsSep " " final.zigBuildFlags} test
+  '';
+
+  postInstall = ''
+    mkdir -p "$terminfo/share"
+    mv "$out/share/terminfo" "$terminfo/share/terminfo"
+    ln -sf "$terminfo/share/terminfo" "$out/share/terminfo"
+
+    mkdir -p "$shell_integration"
+    mv "$out/share/ghostty/shell-integration" "$shell_integration/shell-integration"
+    ln -sf "$shell_integration/shell-integration" "$out/share/ghostty/shell-integration"
+
+    mv "$out/share/vim/vimfiles" "$vim"
+    ln -sf "$vim" "$out/share/vim/vimfiles"
+  '';
+
+  postFixup = ''
+    patchelf --add-rpath "${lib.makeLibraryPath [ libX11 ]}" "$out/bin/.ghostty-wrapped"
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://ghostty.org/";
+    description = "Fast, native, feature-rich terminal emulator pushing modern features.";
+    longDescription = ''
+      Ghostty is a terminal emulator that differentiates itself by being
+      fast, feature-rich, and native. While there are many excellent terminal
+      emulators available, they all force you to choose between speed,
+      features, or native UIs. Ghostty provides all three.
+    '';
+    downloadPage = "https://ghostty.org/download";
+
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
```

```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

If it doesn’t build on macOS, it would be better to set `broken`

---

It's less that it doesn't build on Darwin and more it *can't* build on Darwin since xcodebuild has to be invoked outside of the Nix build environment. I think we should just repackage the .dmg for now until we somehow figure out a way to compile it with Nix

---

I’m curious why that’s the case

And whether that’s been tested since the darwin SDK refactor

---

Ultimately neither I nor @jcollie have Darwin setups to test with, so either someone comes forward with a better idea, or we have to stick with a repackaging for now

---

I don't think that anyone has tried since the darwin refactor. _I_ won't be doing that work because I don't have a macOS machine to test on.

---

https://github.com/ghostty-org/ghostty/issues/1633 is a relevant issue. And https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/.github/workflows/test.yml#L147-L181 is the current CI. Perhaps you can figure out the darwin specific changes as a follow on PR @Enzime?

---

If we want to try and build this with `xcodebuild` we can do what the MacVim derivation does, which punches a hole in the macOS sandbox so it can run `xcodebuild`. Though of course if we can get it built without that it would be much better.

---

I think it makes more sense for the moment to repackage the DMG. We do that for other packages, for example, MS Teams.
I suggest this because it's cleaner than punching holes into the sandbox, and it's less likely to have side effects.
Though it is quite interesting that according to the link @anund shared, it seems to build.

---

> I don't think that anyone has tried since the darwin refactor. _I_ won't be doing that work because I don't have a macOS machine to test on.

I could test I willing be happy to help to support darwin.

---

I believe discussions on this topic are happening on matrix https://matrix.to/#/#macos:nixos.org

---

Bit of a drive by comment but: for my setup and https://github.com/kovidgoyal/kitty I eventually gave up and used the dmg directly; some features (like OS notifications) require signed executables

https://github.com/nmattia/homies/blob/1c27051828b7c10ba2cda8cc790cc50bd796b423/kitty/default.nix#L11

---

Please keep the macOS discussion for follow-up PRs or issues. @jcollie has already mentioned that they won't be the one doing any sort of macOS support.

Please add the platforms though and set badPlatforms or broken.